drivers/unix/os_unix.cpp

```diff
@@ -179,8 +178,18 @@ Vector<String> OS_Unix::get_video_adapter_driver_info() const {
 }
 
 String OS_Unix::get_stdin_string() {
-	char buff[1024];
-	return String::utf8(fgets(buff, 1024, stdin));
+	String result;
+	ssize_t size = BUFSIZ;
+	LocalVector<char> buff;
+	int flags = fcntl(STDIN_FILENO, F_GETFL, 0);
+	if (flags != -1) {
+		buff.resize(size);
+		while ((size = read(fcntl(STDIN_FILENO, F_SETFL, flags | O_NONBLOCK), &buff[0], size)) > 0) {
+			buff.resize(size);
+			result += String::utf8((char *)buff.ptr(), size);
```

Comment:
That is the same as before, and may have issues with cutting in the middle multibyte of character, should like:

```
while (.... read ....) {
     append to buff
}
return String::utf8((char *)buff.ptr()); // after the loop!
```

Reply:
i guess what i am not clear on is the how the custom `LocalVector` type works. After looking at the implementation, it's pretty different from `std::vector` and similar can be said about `String`; `String` is pretty different from `std::string` and `std::wstring`. After reading Godot's documentation, there is no `String.append(str, size)`-like thing available like we have in `std::string` and `std::wstring`, which means i just use `String += whatever` instead. However I haven't been able to find much, if any documentation on `LocalVector` anywhere on Google. To append to a `LocalVector`, is it just a matter of using `+=` like in the case of appending to a `String`?

Reply:
Also, when you confirm this pull request is done being reviewed, I'm going to build the the associated branch of godot from source to do last minute tests just in case the CI might've missed something this pull request might break.

Reply:
The problem here is that the loaded data might end on `C2` for example, and the first byte in the next batch is `A3`, so the first part can't be processed properly, the issue isn't appending to the string, it's that the input isn't guaranteed to be valid utf8 each time

Reply:
`LocalVector<char>` is just an array of bytes, it doesn't case about content. `String` is `Vector<char32_t>`, so it should be converted from UTF-8 to UTF-32. The issue is how UTF-8 encoding and `String::utf8` method works, UTF-8 character can be 1 to 6 bytes long, and `utf8` function will be able to decode only the whole characters. And `read` does not guarantee that character will be read in full. So in case only some single character's bytes are in the current buffer and some are in the next one, `utf8` will fail for both parts. You need to ensure this won't happen: either read everything to the single buffer and that convert to String, or analyze UTF-8 sequences in place and keep cut parts for the next buffer (which is an unnecessary complexity in this case).

Reply:
I sorry for being a bit dense, but while i know how to do these things in C++ using my own code, I don't know how to achieve it with Godot-isms like the `String`, `LocalVector`, and `Vector` types. I might've made this pull request a bit prematurely as either I will need non-pseudo code ("actual code") handed to me to get a better understanding of what to do here, or i might be faced with countless hours of research and debugging. Either is fine, but i find if someone else handled this pull request, thing probably would've gone done more quickly (and smoothly).